Added recipes for my Emacs themes

madara123pain opened the pull request:
### Brief summary of what the package does

This pull request adds recipes for my custom Emacs themes:

- **Amber Glow Theme**
- **Berry Theme**
- **Ember Twilight Theme**
- **Marron Gold Theme**
- **Roseline Theme**
- **Solarized Gruvbox Theme**
- **Spider-Man Theme**

These themes provide a unique color scheme and enhanced UI experience for Emacs users.

### Direct link to the package repository

[https://github.com/madara123pain/unique-emacs-theme-pack](https://github.com/madara123pain/unique-emacs-theme-pack)

### Your association with the package

I am the maintainer and creator of these themes.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

<!-- Please confirm by replacing `[ ]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

madara123pain commented:
I have made the repository public. Let me know if any further changes are needed

madara123pain commented:
> Some of the file names begin with a capital letter but the names of the respective packages do not. You also have accidentally checked in `#Amber-glow-theme.el#`.

I’ve updated the PR by fixing the file naming issues and removing unnecessary files as requested.

riscy commented:
Thanks for the update.  In the future, we suggest sticking to one recipe per PR, especially when you're up to 7 PRs (even though these are related).  Small batches are always easier for everyone - it helps balance our load and lets you prioritize review order.

Now that the recipes are all valid, note the following for each of them:
- Please specify `:fetcher` before `:repo` in your recipe

And beyond that -

⸺ `spider-man-theme.el` with byte-compile using Emacs 30.1:
```
spider-man-theme.el:1:1: Warning: file has no `lexical-binding' directive on its first line
```

⸺ `spider-man-theme.el` with package-lint 20250228.1322:
```
1 issue found:
1:0: warning: The package summary should start with an uppercase letter or a digit.
```

⸺ `spider-man-theme.el` with checkdoc 0.6.2 (fix *within reason*):
```
spider-man-theme.el:30: First sentence should end with punctuation
spider-man-theme.el:30: First line should be capitalized
```

⸺ `solarized-gruvbox-theme.el` with byte-compile using Emacs 30.1:
```
solarized-gruvbox-theme.el:1:1: Warning: file has no `lexical-binding' directive on its first line
```

⸺ `solarized-gruvbox-theme.el` with checkdoc 0.6.2 (fix *within reason*):
```
solarized-gruvbox-theme.el:30: First sentence should end with punctuation
```

- Unreachable package URL in solarized-gruvbox-theme.el: https://github.com/yourusername/solarized-gruvbox-theme

⸺ `roseline-theme.el` with checkdoc 0.6.2 (fix *within reason*):
```
roseline-theme.el:31: First sentence should end with punctuation
```

⸺ `ember-twilight-theme.el` with checkdoc 0.6.2 (fix *within reason*):
```
ember-twilight-theme.el:33: First sentence should end with punctuation
```

⸺ `berry-theme.el` with checkdoc 0.6.2 (fix *within reason*):
```
berry-theme.el:33: First sentence should end with punctuation
```

⸺ `amber-glow-theme.el` with package-lint 20250228.1322:
```
1 issue found:
1:0: warning: Including "Emacs" in the package summary is usually redundant.
```

madara123pain commented:
> Thanks for the update. In the future, we suggest sticking to one recipe per PR, especially when you're up to 7 PRs (even though these are related). Small batches are always easier for everyone - it helps balance our load and lets you prioritize review order.
> 
> Now that the recipes are all valid, note the following for each of them:
> 
>     * Please specify `:fetcher` before `:repo` in your recipe
> 
> 
> And beyond that -
> 
> ⸺ `spider-man-theme.el` with byte-compile using Emacs 30.1:
> 
> ```
> spider-man-theme.el:1:1: Warning: file has no `lexical-binding' directive on its first line
> ```
> 
> ⸺ `spider-man-theme.el` with package-lint 20250228.1322:
> 
> ```
> 1 issue found:
> 1:0: warning: The package summary should start with an uppercase letter or a digit.
> ```
> 
> ⸺ `spider-man-theme.el` with checkdoc 0.6.2 (fix _within reason_):
> 
> ```
> spider-man-theme.el:30: First sentence should end with punctuation
> spider-man-theme.el:30: First line should be capitalized
> ```
> 
> ⸺ `solarized-gruvbox-theme.el` with byte-compile using Emacs 30.1:
> 
> ```
> solarized-gruvbox-theme.el:1:1: Warning: file has no `lexical-binding' directive on its first line
> ```
> 
> ⸺ `solarized-gruvbox-theme.el` with checkdoc 0.6.2 (fix _within reason_):
> 
> ```
> solarized-gruvbox-theme.el:30: First sentence should end with punctuation
> ```
> 
>     * Unreachable package URL in solarized-gruvbox-theme.el: https://github.com/yourusername/solarized-gruvbox-theme
> 
> 
> ⸺ `roseline-theme.el` with checkdoc 0.6.2 (fix _within reason_):
> 
> ```
> roseline-theme.el:31: First sentence should end with punctuation
> ```
> 
> ⸺ `ember-twilight-theme.el` with checkdoc 0.6.2 (fix _within reason_):
> 
> ```
> ember-twilight-theme.el:33: First sentence should end with punctuation
> ```
> 
> ⸺ `berry-theme.el` with checkdoc 0.6.2 (fix _within reason_):
> 
> ```
> berry-theme.el:33: First sentence should end with punctuation
> ```
> 
> ⸺ `amber-glow-theme.el` with package-lint 20250228.1322:
> 
> ```
> 1 issue found:
> 1:0: warning: Including "Emacs" in the package summary is usually redundant.
> ```

Thanks for the review! 🙌 I’ve made the following fixes based on your feedback:

    Moved :fetcher before :repo in all recipes.
    Added lexical-binding to all theme files.
    Fixed punctuation and capitalization in package summaries.
    Corrected the URL issue in solarized-gruvbox-theme.el.
    Removed "Emacs" from the package summary where redundant.

Please let me know if anything else needs adjustment. Thanks again for your time!

madara123pain commented:
Thanks for the review! 🙌 I’ve made the following fixes based on your feedback:

    Moved :fetcher before :repo in all recipes.
    Added lexical-binding to all theme files.
    Fixed punctuation and capitalization in package summaries.
    Corrected the URL issue in solarized-gruvbox-theme.el.
    Removed "Emacs" from the package summary where redundant.

Please let me know if anything else needs adjustment. Thanks again for your time!

riscy commented:
I will merge these but I wanted to mention a couple of things:

Your screenshot pngs are all on the same branch as your themes.  This means MELPA, but also people installing through tools like straight.el, will have to download all of your png files as well as the source files.  You can help everyone out if you e.g. put your screenshots on a separate branch.  Refer to how e.g. [doom-themes](https://github.com/doomemacs/themes) does it.

Second: the suitability of a given theme is in the eye of the beholder, and so I'm usually not very picky at all (but maybe I should start).  But I suggest spending some times with this set of themes before writing more.  Quality over quantity.  Do you actually use these?  Do you actually use berry-theme? :)  Also, each one of these is a very "limited" theme (less than 100 definitions, where you often see several hundreds in established emacs themes, covering many modes).

madara123pain commented:
> I will merge these but I wanted to mention a couple of things:
> 
> Your screenshot pngs are all on the same branch as your themes. This means MELPA, but also people installing through tools like straight.el, will have to download all of your png files as well as the source files. You can help everyone out if you e.g. put your screenshots on a separate branch. Refer to how e.g. [doom-themes](https://github.com/doomemacs/themes) does it.
> 
> Second: the suitability of a given theme is in the eye of the beholder, and so I'm usually not very picky at all (but maybe I should start). But I suggest spending some times with this set of themes before writing more. Quality over quantity. Do you actually use these? Do you actually use berry-theme? :) Also, each one of these is a very "limited" theme (less than 100 definitions, where you often see several hundreds in established emacs themes, covering many modes).

For the PNGs – I didn’t know about that, but I’ll rearrange them as suggested.

As for theme quality – I don’t use berry-theme, but I do use my other themes. I made it because it’s unique, but I understand the importance of quality over quantity.

Regarding limited definitions – I get it. In my next update, I’ll try to cover more modes for better support.
